test-runner-poc/environments.js

```diff
@@ -0,0 +1,267 @@
+'use strict';
+
+const f = require('util').format;
+const semver = require('semver');
+const path = require('path');
+const core = require('../lib/core');
+
+/**
+ * Base class for environments in projects that use the test
+ * runner
+ */
+class EnvironmentBase {
+  /**
+   * The default implementation of the environment setup
+   *
+   * @param {*} callback
+   */
+  setup(callback) {
+    callback();
+  }
+}
+
+const genReplsetConfig = (port, options) => {
+  return Object.assign(
+    {
+      options: {
+        bind_ip: 'localhost',
+        port: port,
+        dbpath: `${__dirname}/../db/${port}`,
+        setParameter: ['enableTestCommands=1']
+      }
+    },
+    options
+  );
+};
+
+function usingUnifiedTopology() {
+  return !!process.env.MONGODB_UNIFIED_TOPOLOGY;
+}
+
+/**
+ *
+ * @param {*} discoverResult
+ */
+class ReplicaSetEnvironment extends EnvironmentBase {
+  constructor(host, port, version) {
+    super();
+
+    this.host = host;
+    this.port = port;
+    this.setName = 'replset';
+    this.url = `mongodb://%s${host}:${port}/integration_tests?rs_name=rs`;
+    this.writeConcernMax = { w: 'majority', wtimeout: 30000 };
+    this.replicasetName = 'rs';
+    this.topology = function(topologyHost, topologyPort, options) {
+      topologyHost = topologyHost || host;
+      topologyPort = topologyPort || port;
+      options = Object.assign({}, options);
+      options.replicaSet = 'rs';
+      options.poolSize = 1;
+      options.autoReconnect = false;
+
+      if (usingUnifiedTopology()) {
+        return new core.Topology([{ topologyHost, topologyPort }], options);
+      }
+
+      return new core.ReplSet([{ topologyHost, topologyPort }], options);
+    };
+
+    this.nodes = [
```

Comment:
changed from hardcoded ports to all the ports sequentially after the port passed in

Reply:
Implemented here:
https://github.com/mongodb/node-mongodb-native/pull/2104/commits/1063f380046176c2b0ea9e7b005b4ada1f572c80